### PR TITLE
Use moment for timeslot calculation

### PR DESCRIPTION
**Pull request by charleyf**
### Description
This is a PR to capture some experimental work I did to switch to using moment.js for timeslot calculation.

Notes:
- Timeslots for hearings are being prototyped in this sprint (03-02-2020).
- We already use and import moment.js in this file ([apparently](https://thecodebarbarian.com/a-practical-guide-to-moment-timezone.html) moment-timezone is a superset of moment).

### CODE REVIEW

File: client/app/hearings/utils.js [hunk 1]
```diff
@@ -489,55 +489,46 @@ export const setTimeSlots = (hearings) => {
...
-      hearingTime: `${hour < 10 ? 0 : ''}${hour}:${incrementAmount}`,
+    const slotTime = moment(startTime, 'HH:mm').add(index, 'hours');
+
+    // This slot is not available (full) if there's a scheduled hearing less than an hour before it.
+    // A 10:45 appointment will:
+    // - Hide a 11:30 slot (it's full, so we return null)
+    // - Show a 10:30 slot (return the timeslot)
+    const slotFull = scheduledHearingTimes.some((scheduledHearingTime) =>
+      slotTime.isAfter(scheduledHearingTime) &&
+        slotTime.diff(scheduledHearingTime, 'minutes') <= 60
```
Comment (sahalliburton) [inline client/app/hearings/utils.js:519]: I think this needs to be `< 60`, so that if a hearing is scheduled at '10:30', then it will still show '11:30'

Comment (charleyf) [reply client/app/hearings/utils.js]: Ah, good catch.
- Added a failing test for that
- Made the failing test pass by changing '<=' to '<'.

File: client/app/hearings/utils.js [hunk 2]
```diff
@@ -489,55 +489,46 @@ export const setTimeSlots = (hearings) => {
...
+      hearingTime: slotTime.format('HH:mm'),
       full: false
     };
   }));
 
   // Transform the values into the available slots
-  const slots = [...availableTimes, ...scheduledHearings].map((slot) => ({
+  const slots = [...availableSlots, ...scheduledHearings].map((slot) => ({
     ...slot,
-    key: `${slot?.externalId}-${slot?.hearingTime}`,
```
Comment (sahalliburton) [inline client/app/hearings/utils.js:540]: This key is needed for the Timeslots component so that React doesnt get duplicate keys

**Review by sahalliburton — CHANGES_REQUESTED** [review]
LGTM, just one small blocking change before its done

File: client/app/hearings/utils.js (hunk 2 above)
Comment (charleyf) [reply client/app/hearings/utils.js:540]: Ah, dang. I thought I'd kept that. Thanks!

Comment (charleyf) [reply client/app/hearings/utils.js:540]: Added this back with a test and extended it slightly, before would get something like this:
'-10:30' (slot)
'aa385-58568-338...-10:45' (scheduled_hearing)
'-12:30' (slot)

Now we will get this:
'2-10:30' (slot)
'aa385-58568-338...-10:45' (scheduled_hearing)
'4-12:30' (slot)

**Review by sahalliburton — APPROVED** [review]

**Pull request merged**